providers/posflag/posflag.go

```diff
@@ -108,6 +100,54 @@ func (p *Posflag) Read() (map[string]interface{}, error) {
 			val = v
 		}
 
+		// if val hasn't been set yet, check if the flag has a Get() method.
+		if val == nil {
+			getter, ok := f.Value.(flag.Getter)
```

Comment:
The flag.Getter interface comes from the standard library. The reason it's not in the flag.Value interface is that it was added after Go 1:

https://pkg.go.dev/flag#Getter
> Getter is an interface that allows the contents of a Value to be retrieved. It wraps the Value interface, rather than being part of it, because it appeared after Go 1 and its compatibility rules.